src/cli/commands/add.js

```diff
@@ -107,7 +107,22 @@ export class Add extends Install {
 
   async init(): Promise<Array<string>> {
     this.addedPatterns = [];
-    const patterns = await Install.prototype.init.call(this);
+    let patterns = [];
+    try {
+      patterns = await Install.prototype.init.call(this);
```

Comment:
I am not a huge fan of the try/catch logic here.
In here you recover from an exception but only once, what happens if it throws again?
I think it is better to not rely on exception mechanism for this feature